사다리게임 step01

hngfu opened the pull request:
### 기본 사다리를 구현했습니다.

hngfu committed:
[사다리게임] 기본 사다리 구현

기본 사다리를 구현했습니다.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -7,4 +7,51 @@
 //
 
 import Foundation
+import GameplayKit
```
여기 GameplayKit 은 왜 import 한건가요?

hngfu replied:
스위프트 개발자 문서에 검색 arc4random을 검색하다가
GKARC4RandomSource를 발견했는데 순간 어떠한 착각을 해서 추가 한것 같습니다.
지금은 삭제했습니다..

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -7,4 +7,51 @@
+    if convertedParticipant >= 3 && convertedLadderHeight >= 2 {
+        return makeLadder(convertedParticipant, convertedLadderHeight)
+    }
+    return "입력하신 수를 확인해 주세요."
```
initialize() 이름은 초기 단계만 책임질 것 같은데, 이 함수가 전체 로직을 다 포괄하는 것 같네요. 
이렇게 최상위 흐름을 제어하는 main()과 같은 함수는 맨 아래에 별도로 작성해보세요.
대신 로직을 최소화하고 다른 함수를 호출해서 리턴값을 다른 함수로 전달하는 정도만 포함하는 게 좋습니다. 
여기 3단락으로 나눠진 부분을 다른 함수로 분리해보세요.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -7,4 +7,51 @@
+//입력: "12" -> 출력: 12(Int), 입력: "asd" -> 출력: 0
+func convertToInt(_ valueToConvert:String) -> Int {
+    guard let convertedValue = Int(valueToConvert) else { return 0 }
+    return convertedValue
```
convertToInt() 함수 이름에 Int 같은 타입을 포함시키지 마세요.
어차파 `func convertToInt(_ valueToConvert:String) -> Int`  리턴타입을 보면 알 수 있는 내용입니다. 과감하게 빼세요.
매개변수 이름에도 `valueToConvert` ToConvert가 또 붙네요. 함수 이름과 같은걸 또 넣는게 중복이지 않을까요?

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -7,4 +7,51 @@
+func makeLadder(_ participant:Int, _ maximumLadderHeight:Int) -> String {
+    var ladder = ""
+    for _ in 0..<maximumLadderHeight {
+        for i in makeHorizontalOfLadder(participant) {
```
for in 뒤에 메소드를 호출하기 보다는 let으로 명확하게 받아서 처리하는 게 더 깔끔합니다. 
ladder 를 "" 빈 문자열이 아니라 2차원 배열로 접근해보세요.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -7,4 +7,51 @@
+    for i in 0..<elementsBetweenBars.count {
+        horizontalOfLadder[(i * 2) + 1] = elementsBetweenBars[i]
+    }
+    horizontalOfLadder.append("\n")
```
"|" horizontal 은 사다리를 출력하기 위한 보조적인 값입니다.
사다리 데이터에 이 값을 포함하지 않고, elementsBetweenBars 에 해당하는 값만 표현하도록 개선해보세요.
대신 출력하는 함수를 아래처럼 print() 하나만 하는게 아니라, 사다리 데이터를 받아서 출력하는 함수를 만들어보세요.

hngfu replied:
이 피드백을 보고 원래는 한층 한층 |-| | <- 이렇게 만들어서 쌓는 방법에서
미리 ||| <- 이렇게 만들어진 사다리에 '-' 또는 ' ' (공백)을 추가하여  만드는 방법으로 수정했는데
jk가 말씀하신게 이게 맞는지 궁금합니다.

hngfu replied:
방금 다시 생각보니 좋은 아이디어가 생각났습니다!
아마 jk가 말씀하신게 이게 아닌가 싶습니다..!
다시 해보겠습니다!!!

godrm reviewed (commented):
다음과 같은 피드백을 확인해서 개선해보세요. 
변경한 내용은 지금 브랜치에서 commit / push 만 하면 됩니다.

hngfu committed:
[사다리게임] 두번째 피드백까지 수정 완료

main()함수 생성과 유효성검사함수 부분 생성

hngfu committed:
[사다리게임] 3번째 피드백 완료

몇몇 함수명 변경

hngfu committed:
[사다리게임] 마지막피드백까지 수정완료

2차원 배열로 변환 완료. 사다리 데이터를 받아서 2중 루프로 출력하도록 만듦.

hngfu commented:
## 수정내용
1. import했던 GameplayKit를 삭제했습니다.
2. main()함수 생성했습니다.
3. 리턴값을 이용하여 값이 이동하도록 했습니다.
4. 최소값 체크 부분을 함수로 만들었습니다.
5. 피드백 주신 함수명과 그 외에 다른 변수들의 이름을 바꿨습니다.
6. 2차원 배열로 바꿨습니다.
7. 사다리 데이터를 받아서 출력하는 함수를 만들었습니다.

hngfu committed:
[사다리게임] 피드백 수정 완료

피드백 내용 전부 수정 완료하였습니다.
손잡이가 완전 랜덤으로 생기도록 구현하였습니다.
손잡이가 연결될 경우는 배제하도록 구현하였습니다.
'|'는 출력할때 그려주도록 만들었습니다.

hngfu commented:
## 수정내용

1. 피드백 내용 전부 수정 완료하였습니다.
2. 손잡이가 완전 랜덤으로 생기도록 구현하였습니다.
3. 손잡이가 연결될 경우는 배제하도록 구현하였습니다.
4. '|'는 출력할때 그려주도록 만들었습니다.

godrm commented:
수고하셨습니다. 첫 단계 요구사항을 만족한 것 같습니다.
- 다음 부터는 commit 하는 단위를 좀 더 세밀하게 나눠보세요. 수정내용 하나씩 커밋 단위가 될 수도 있습니다. 
- 커밋 로그도 피드백 내용을 수정하는게 아니라 어떤 코드에 어떤 내용을 어떻게 작업했는지 작업에 대해 남겨보세요. 로그만 보고 변경 사항을 찾을 수 있어야 좋습니다. 좋은 커밋 로그에 대해서 공부해보세요.
